src/components/posts/index.js

```diff
@@ -1,27 +1,49 @@
 import { useEffect, useState } from "react";
 import Post from "../post";
-import { getPosts } from "../../service/apiClient";
 
 const Posts = () => {
-    const [posts, setPosts] = useState([])
+  const [posts, setPosts] = useState([]);
 
-    useEffect(() => {
-        getPosts().then(setPosts)
-    }, [])
+  useEffect(() => {
+    const token = localStorage.getItem('token'); // Adjust according to where your token is stored
 
-    return (
-        <>
-            {posts.map(post => {
-                    return <Post
-                        key={post.id}
-                        name={`${post.author.firstName} ${post.author.lastName}`}
-                        date={post.createdAt}
-                        content={post.content}
-                        comments={post.comments}
-                    />
-            })}
-        </>
-    )
-}
+    fetch('http://localhost:4000/posts', {
+        headers: {
+            'Authorization': `Bearer ${token}` // Include the authorization header
+        }
+    })
+    .then(response => {
+        if (!response.ok) {
+            throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+        return response.json();
+    })
+    .then(data => {
+        // Correctly accessing the nested structure of the response
+        if (data.data && data.data.posts && Array.isArray(data.data.posts)) {
```

Comment:
Let's flip this around. So instead check if the data is not there / is not an array. If that is the case, throw an error. Otherwise continue as normal

```js
if (!data.data || !data.data.posts || !Array.isArray(data.data.posts)) {
   throw new Error
}
```